core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -266,16 +265,15 @@ sealed abstract class IO[E, A] { self =>
    * error to the cleanup action. The cleanup action will not be interrupted.
    * Cleanup actions for handled and unhandled errors can be provided separately.
    */
-  final def onError(cleanupT: Throwable => Infallible[Unit])(cleanupE: E => Infallible[Unit]): IO[E, A] =
-    IO.unit[E]
-      .bracket0(
-        (r: ExitResult[E, A], a: Unit) =>
-          r match {
-            case ExitResult.Failed(e)     => cleanupE(e)
-            case ExitResult.Terminated(t) => cleanupT(t)
-            case _                        => IO.unit
-        }
-      )(_ => self)
+  final def onError(cleanupE: Option[E] => Infallible[Unit]): IO[E, A] =
```

Comment:
Scaladocs got outdated after this change. Also `cleanupE` could be renamed to just `cleanup`